src/hotspot/share/sanitizers/ub.hpp

```diff
@@ -33,7 +33,10 @@
 // Useful if the function or method is known to do something special or even 'dangerous', for
 // example causing desired signals/crashes.
 #ifdef UNDEFINED_BEHAVIOR_SANITIZER
-#if defined(__clang__) || defined(__GNUC__)
+#if defined(__clang__)
+#define ATTRIBUTE_NO_UBSAN __attribute__((no_sanitize("undefined","float-divide-by-zero")))
+#endif
+#if defined(__GNUC__) && !defined(__clang__)
```

Comment:
Maybe use `#elif`?

Reply:
Or should I try to set it for BOTH clang and gcc so that the added checks are not needed any more?

Reply:
If you change it to #elif, then please do it in both files.